tsdb/db.go

```diff
@@ -1484,7 +1489,9 @@ func (db *DB) compactBlocks() (err error) {
 		default:
 		}
 
+		db.mtx.RLock()
 		uids, err := db.compactor.Compact(db.dir, plan, db.blocks)
+		db.mtx.RUnlock()
```

Comment:
Compact calls `compactOOOHead` which calls `db.mtx.Lock()`; won't that deadlock?

Reply:
Possibly, that lock was added in 59844498f7b12f16c7f004aa951bbb14cdb83991, after this PR was created, so I would have to re-investigate all calls

Reply:
And this lock here was only added after this comment - https://github.com/prometheus/prometheus/pull/12920#pullrequestreview-1684220638
So need to re-check that part

Reply:
I think this RLock/RUnlock might have been added incorrectly during rebasing, I don't see why it's needed there